File: front-end/src/service/forumService.js
import axios from "./api";

/** Classe service para os fóruns */
class ForumService {

    /** Função para buscar todos os fóruns salvos */
    async getAll() {
        return (await axios.get(`/forum`, { withCredentials: true })).data;
    }
}

export default new ForumService();

File: back-end-maven/src/main/java/net/weg/wegssm/model/service/ForumService.java
package net.weg.wegssm.model.service;

import lombok.AllArgsConstructor;
import net.weg.wegssm.model.entities.Forum;
import net.weg.wegssm.repository.ForumRepository;
import org.springframework.stereotype.Service;

import java.util.List;

/**
 * Classe service para os fóruns
 */
@Service
@AllArgsConstructor
public class ForumService {

    /**
     * Classe repository dos fóruns
     */
    ForumRepository forumRepository;

    /**
     * Função para buscar todos os fóruns salvos no banco de dados
     */
    public List<Forum> findAll() {
        return forumRepository.findAll();
    }

}
